src/core/namespace.c

```diff
@@ -787,6 +796,32 @@ static int prefix_where_needed(MountList *ml, const char *root_directory) {
         return 0;
 }
 
+static bool verity_has_duplicates(MountList *ml, const MountEntry *needle) {
+
+        assert(ml);
+        assert(needle);
+
```

Comment:
i think we should validate that `needle` is MOUNT_EXTENSION_IMAGE.

either by returning false if that's not the case (after all non-extension-image mounts never have relevant duplicates). Or via an assert()